include/Hacl_RSAPSS.h

```diff
@@ -42,10 +42,12 @@ extern "C" {
 /**
 Sign a message `msg` and write the signature to `sgnt`.
 
+@note Memory location of `sgnt` must not overlap memory locations of: `skey`, `salt`, and `msg`.
```

Comment:
Just an example of one of the two @notes added to warn users of the disjoint memory pre-condition. This condition applies to many other parameters but is not sensible for a client to attempt memory overlapping in those cases. It seems slightly plausible for a client to attempt overlapping memory with `sgnt`, however, so I decided to only mention the requirement for the two functions that take `*sgnt` and write to it. Thoughts?

Reply:
Could we add a blanket statement like "all memory must be disjoint"?

Reply:
Of course, we could. Although I don't remember seeing that _all_ memory needs to be disjoint, but to assure that the verification guarantees will hold for the client, we can make that blanket reasonable request and simplify things for everyone.

And in that case, the blanket statement should just appear once at the top of the RSAPSS documentation page (which i can't find for hacl-star, by the way) before the function signatures.